Stack/Opc.Ua.Core/Stack/Server/ServerBase.cs

```diff
@@ -778,6 +778,40 @@ public static bool RequireEncryption(EndpointDescription description)
             return requireEncryption;
         }
 
+        /// <summary>
+        /// Sets the Server Certificate in an Endpoint description if the description requires encryption
+        /// </summary>
+        /// <param name="description">the endpoint Description to set the server certificate</param>
+        /// <param name="sendCertificateChain">true if the certificate chain shall be sent</param>
+        /// <param name="instanceCertificate">the instance certificate</param>
+        /// <param name="instanceCertificateChain">the instance certificate chain</param>
+        public static void SetServerCertificateInEndpointDescription(
+            EndpointDescription description,
+            bool sendCertificateChain,
+            X509Certificate2 instanceCertificate,
+            X509Certificate2Collection instanceCertificateChain)
+        {
+            if (RequireEncryption(description))
+            {
+                description.ServerCertificate = instanceCertificate.RawData;
+
+                // check if complete chain should be sent.
+                if (sendCertificateChain &&
+                    instanceCertificateChain != null &&
+                    instanceCertificateChain.Count > 1)
+                {
+                    List<byte> serverCertificateChain = new List<byte>();
```

Comment:
pre allocate the array and copy rawdata to it?

Reply:
```C#
    int totalSize = instanceCertificateChain.Sum(cert => cert.RawData.Length);
    List<byte> serverCertificateChain = new List<byte>(totalSize);
```

like this? , is this really a performance benefit?

Reply:
```C#
    int totalSize = instanceCertificateChain.Sum(cert => cert.RawData.Length);
    var serverCertificateChain = new byte[totalSize];
    int offset = 0;
    foreach (var cert in instanceCertificateChain)
    {
        Array.Copy(cert.RawData, 0, serverCertificateChain, offset, cert.RawData.Length);
        offset +=  cert.RawData.Length;
    }
```